src/ert/job_queue/queue.py

```diff
@@ -97,24 +97,15 @@ def __repr__(self) -> str:
     def __str__(self) -> str:
         return self.__repr__()
 
-    def __init__(self, driver: "Driver", max_submit: int = 2):
-        """
-        Short doc...
-        The @max_submit argument says how many times the job should be submitted
-        (including a failure)
-              max_submit = 2: means that we can submit job once more
-        The @size argument is used to say how many jobs the queue will
-        run, in total.
-        """
-
+    def __init__(self, queue_config: QueueConfig):
         self.job_list: List[JobQueueNode] = []
         self._stopped = False
-        c_ptr = self._alloc(driver.from_param(driver))
+        self.driver: Driver = Driver.create_driver(queue_config)
+        c_ptr = self._alloc(self.driver.from_param(self.driver))
         super().__init__(c_ptr)
 
-        self.driver = driver
         self._differ = QueueDiffer()
-        self._max_submit = max_submit
+        self._max_submit = queue_config.max_submit if queue_config.max_submit else 2
```

Comment:
Is this already a default value created in the queue_config dataclass?

Reply:
I think you are right! the if-else-thing is now removed, and the dataclass default is used instead.